brain/section.h

```diff
@@ -0,0 +1,135 @@
+ * FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#ifndef BRAIN_SECTION
+#define BRAIN_SECTION
+
+#include <brain/api.h>
+#include <brain/types.h>
+
+namespace brain
+{
+
+namespace detail
+{
+class Morphology;
+}
+
+/**
+ * A class to represent a morphological section.
+ *
+ * A Section is an unbranched piece of a morphological skeleton.
+ * This class provides functions to query information about the sample points
+ * that conform the section and functions to obtain the parent and children
+ * sections.
+ *
+ * The cell soma is also considered a section, but some functions have
+ * special meaning for it.
+ *
+ * Sections cannot be directly created, but are returned by several
+ * brain::Morphological and brain::Section methods.
+ *
+ * This is a lightweight object with STL container style thread safety.
+ * It is also safe to use a section after the morphology from where it comes
+ * has been deallocated. The morphological data will be keept as long as there
+ * is a Section referring to it.
+ */
+class Section
+{
+public:
+    friend class Morphology;
+
+    Section( const Section& section );
+
+    ~Section();
+
+    Section& operator=( const Section& section );
+
+    bool operator==( const Section& section ) const;
+
+    /** Return the ID of this section. */
+    uint32_t getID() const;
+
+    SectionType getType() const;
+
+    /**
+     * Return the total length of this section.
+     *
+     * If this section is a soma section the length is ill-defined and this
+     * function will return 0.
+     */
+    BRAIN_API float getLength() const;
+
+    /**
+     * Return the list of all point samples that define this section.
+     *
+     * If this sections is a soma section return the list of points in the
+     * soma ring.
+     *
+     * @return A list of point positions with radius. For a section consisting
+     *         of n segments, this list will have n + 1 points.
+     */
+    BRAIN_API Vector4fs getSamples() const;
+
+    /**
+     * Return a list of points sampling this section at discrete locations.
+     *
+     * If the section is a soma section this function will return the soma
+     * position for all sampling positions. The soma position is assumed to
+     * be (0, 0, 0) unless the origin morphology has been transformed.
+     *
+     * @param points Normalized positions of the sample points along the
+     *        section. Values will be clampled to [0, 1] before sampling.
+     * @return The vector with all the points of the section if points is empty.
+     *         The section sampled at the given points otherwise.
+     */
+    BRAIN_API Vector4fs getSamples( const floats& points ) const;
```

Comment:
No. First it would be getSamples( floats( )). Then, getSamples( points ) returns samples at specific locations, getSamples( floats( )) would return an empty list. On the other hand getSamples() returns the sampling positions as they are.